packages/contracts-bedrock/src/universal/CrossDomainMessenger.sol

```diff
@@ -266,16 +271,17 @@ abstract contract CrossDomainMessenger is
         // If there is not enough gas left to perform the external call and finish the execution,
         // return early and assign the message to the failedMessages mapping.
         // We are asserting that we have enough gas to:
-        // 1. Call the target contract (_minGasLimit + RELAY_CALL_OVERHEAD + RELAY_GAS_CHECK_BUFFER)
+        // 1. Call the target contract (_minGasLimit + RELAY_CALL_OVERHEAD + RELAY_GAS_CHECK_BUFFER +
+        // _relayMessageValidatorConfigGas)
         //   1.a. The RELAY_CALL_OVERHEAD is included in `hasMinGas`.
+        //   1.b. The relay message validation config check is included in `hasMinGas`.
         // 2. Finish the execution after the external call (RELAY_RESERVED_GAS).
         //
         // If `xDomainMsgSender` is not the default L2 sender, this function
         // is being re-entered. This marks the message as failed to allow it to be replayed.
         if (
             !SafeCall.hasMinGas(
-                _minGasLimit,
-                RELAY_RESERVED_GAS + RELAY_GAS_CHECK_BUFFER + _relayMessageValidationGas(uint64(_message.length))
+                _minGasLimit, RELAY_RESERVED_GAS + RELAY_GAS_CHECK_BUFFER + _relayMessageValidatorConfigGas()
```

Comment:
This isn't perfect, but you could do something like 
```solidity

RELAY_RESERVED_GAS + RELAY_GAS_CHECK_BUFFER + _relayMessageValidatorConfigGas() + (relayMessageValidator == address(0) ? 0 : _relayMessageValidationGas(_message.length))

```

Reply:
commented about that here - https://github.com/Firewall-eng/optimism/pull/5/files#r1632120366